src/main/java/com/google/swarm/tokenization/common/CSVReader.java

```diff
@@ -0,0 +1,110 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.swarm.tokenization.common;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import org.apache.beam.sdk.io.FileIO.ReadableFile;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.values.KV;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.swarm.tokenization.CSVStreamingPipeline;
+
+@SuppressWarnings("serial")
+public class CSVReader extends DoFn<ReadableFile, KV<String, List<String>>> {
+	public static final Logger LOG = LoggerFactory.getLogger(CSVStreamingPipeline.class);
+	private ValueProvider<String> csek;
+	private ValueProvider<String> csekHash;
+	private ValueProvider<String> fileDecryptKeyName;
+	private ValueProvider<String> fileDecryptKey;
+	private String projectId;
+	private ValueProvider<Integer> batchSize;
+
+	public CSVReader(ValueProvider<String> csek, ValueProvider<String> csekHash,
+			ValueProvider<String> fileDecryptKeyName, ValueProvider<String> fileDecryptKey, String projectId,
+			ValueProvider<Integer> batchSize) {
+		this.csek = csek;
+		this.csekHash = csekHash;
+		this.fileDecryptKeyName = fileDecryptKeyName;
+		this.fileDecryptKey = fileDecryptKey;
+		this.projectId = projectId;
+		this.batchSize = batchSize;
+
+	}
+
+	private static <T> Collection<List<T>> partition(List<T> list, int size) {
+		final AtomicInteger counter = new AtomicInteger(0);
+
+		return list.stream().collect(Collectors.groupingBy(it -> counter.getAndIncrement() / size)).values();
+	}
+
+	@ProcessElement
+	public void processElement(ProcessContext c) {
+		ReadableFile f = c.element();
+		boolean customerSuppliedKey = false;
+		String key = null;
+		BufferedReader br = null;
+		List<String> contents = new ArrayList<>();
+
+		if (this.csek.isAccessible()) {
+
+			customerSuppliedKey = Util.findEncryptionType(this.fileDecryptKeyName.get(), this.fileDecryptKey.get(),
+					this.csek.get(), this.csekHash.get());
+
+		}
+
+		if (customerSuppliedKey) {
+
+			try {
+				key = KMSFactory.decrypt(this.projectId, "global", this.fileDecryptKeyName.get(),
+						this.fileDecryptKey.get(), this.csek.get());
+			} catch (IOException e) {
+
+				e.printStackTrace();
+			} catch (GeneralSecurityException e) {
+
+				e.printStackTrace();
+			}
+		}
+		String bucketName = Util.parseBucketName(f.getMetadata().resourceId().getCurrentDirectory().toString());
+
+		String objectName = f.getMetadata().resourceId().getFilename().toString();
+		String[] fileKey = objectName.split("\\.", 2);
```

Comment:
Does splitting on `"\\." `work for getting the fileKey?

Reply:
Yes. Did you find any issue?